pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/qt68.patch

```diff
@@ -0,0 +1,355 @@
+diff --git a/Telegram/SourceFiles/data/data_document.cpp b/Telegram/SourceFiles/data/data_document.cpp
+index 9c25a7861..5456ef280 100644
+--- a/Telegram/SourceFiles/data/data_document.cpp
++++ b/Telegram/SourceFiles/data/data_document.cpp
+@@ -936,14 +936,14 @@ void DocumentData::setFileName(const QString &remoteFileName) {
+ 	// in filenames, because they introduce a security issue, when
+ 	// an executable "Fil[x]gepj.exe" may look like "Filexe.jpeg".
+ 	QChar controls[] = {
+-		0x200E, // LTR Mark
+-		0x200F, // RTL Mark
+-		0x202A, // LTR Embedding
+-		0x202B, // RTL Embedding
+-		0x202D, // LTR Override
+-		0x202E, // RTL Override
+-		0x2066, // LTR Isolate
+-		0x2067, // RTL Isolate
++		QChar(0x200E), // LTR Mark
++		QChar(0x200F), // RTL Mark
++		QChar(0x202A), // LTR Embedding
++		QChar(0x202B), // RTL Embedding
++		QChar(0x202D), // LTR Override
++		QChar(0x202E), // RTL Override
++		QChar(0x2066), // LTR Isolate
++		QChar(0x2067), // RTL Isolate
+ 	};
+ 	for (const auto &ch : controls) {
+ 		_filename = std::move(_filename).replace(ch, "_");
+diff --git a/Telegram/SourceFiles/data/data_wall_paper.cpp b/Telegram/SourceFiles/data/data_wall_paper.cpp
+index 1bfd65335..0234bdbb4 100644
+--- a/Telegram/SourceFiles/data/data_wall_paper.cpp
++++ b/Telegram/SourceFiles/data/data_wall_paper.cpp
+@@ -150,8 +150,8 @@ using Ui::MaybeColorFromSerialized;
+ 		const auto hex = [](int value) {
+ 			value = std::clamp(value, 0, 15);
+ 			return (value > 9)
+-				? ('a' + (value - 10))
+-				: ('0' + value);
++				? QChar('a' + (value - 10))
++				: QChar('0' + value);
+ 		};
+ 		return QString() + hex(value / 16) + hex(value % 16);
+ 	};
+diff --git a/Telegram/SourceFiles/lang/lang_cloud_manager.cpp b/Telegram/SourceFiles/lang/lang_cloud_manager.cpp
+index 8c98b96d1..d7692e0be 100644
+--- a/Telegram/SourceFiles/lang/lang_cloud_manager.cpp
++++ b/Telegram/SourceFiles/lang/lang_cloud_manager.cpp
+@@ -434,7 +434,6 @@ void CloudManager::requestLanguageAndSwitch(
+ 
+ void CloudManager::sendSwitchingToLanguageRequest() {
+ 	if (!_api) {
+-		_switchingToLanguageId = -1;
```

Comment:
The `QChar` constructs are self-explanatory, but what's with this line?

Reply:
This is a QString, which used to be initialized via  implicit QString(QChar(int)). I have no idea what the intent was, but thankfully the value isn't actually ever used.